src/mlpack/methods/random_forest/random_forest_regressor_impl.hpp

```diff
@@ -0,0 +1,510 @@
+         const arma::Row<ResponsesType>& responses,
+         const arma::rowvec& weights,
+         const size_t numTrees,
+         const size_t minimumLeafSize,
+         const double minimumGainSplit,
+         const size_t maximumDepth,
+         const bool warmStart,
+         DimensionSelectionType dimensionSelector)
+{
+  // Pass off to Train().
+  return Train<true, true>(dataset, datasetInfo, responses, weights,
+      numTrees, minimumLeafSize, minimumGainSplit, maximumDepth,
+      dimensionSelector, warmStart);
+}
+
+template<
+    typename FitnessFunction,
+    typename DimensionSelectionType,
+    template<typename> class NumericSplitType,
+    template<typename> class CategoricalSplitType,
+    bool UseBootstrap
+>
+template<typename VecType>
+double RandomForestRegressor<
+    FitnessFunction,
+    DimensionSelectionType,
+    NumericSplitType,
+    CategoricalSplitType,
+    UseBootstrap
+>::Predict(const VecType& point) const
+{
+   // Check edge case.
+  if (trees.size() == 0)
+  { 
+    throw std::invalid_argument("RandomForestRegressor::Predict(): no random forest "
+        "trained!");
+    return 0;
+  }
+
+  double prediction;
+  double totalSum = 0;
+  for (size_t i = 0; i < trees.size(); ++i)
+  {
+    totalSum += trees[i].Predict(point);
+  }
+
+  // Find maximum element after renormalizing probabilities.
+  prediction = totalSum / trees.size();
+
+  return prediction;
+}
+
+template<
+    typename FitnessFunction,
+    typename DimensionSelectionType,
+    template<typename> class NumericSplitType,
+    template<typename> class CategoricalSplitType,
+    bool UseBootstrap
+>
+template<typename MatType, typename ResponsesType>
+void RandomForestRegressor<
+    FitnessFunction,
+    DimensionSelectionType,
+    NumericSplitType,
+    CategoricalSplitType,
+    UseBootstrap
+>::Predict(const MatType& data,
+            arma::Row<ResponsesType>& predictions) const
+{
+  // Check edge case.
+  if (trees.size() == 0)
+  {
+    predictions.clear();
+
+    throw std::invalid_argument("RandomForestRegressor::Predict(): no random forest "
+        "trained!");
+  }
+
+  predictions.set_size(data.n_cols);
+
+  #pragma omp parallel for
+  for (size_t i = 0; i < data.n_cols; ++i)
+  {
+    predictions[i] = Predict(data.col(i));
+  }
+}
+
+template<
+    typename FitnessFunction,
+    typename DimensionSelectionType,
+    template<typename> class NumericSplitType,
+    template<typename> class CategoricalSplitType,
+    bool UseBootstrap
+>
+template<typename Archive>
+void RandomForestRegressor<
+    FitnessFunction,
+    DimensionSelectionType,
+    NumericSplitType,
+    CategoricalSplitType,
+    UseBootstrap
+>::serialize(Archive& ar, const uint32_t /* version */)
+{
+  size_t numTrees;
+  if (cereal::is_loading<Archive>())
+    trees.clear();
+  else
+    numTrees = trees.size();
+
+  ar(CEREAL_NVP(numTrees));
+
+  // Allocate space if needed.
+  if (cereal::is_loading<Archive>())
+    trees.resize(numTrees);
```

Comment:
I think all of this is actually a relic from older serialization code that did not work correctly with `std::vector`.  So I think it can be removed---the `ar(CEREAL_NVP(trees))` should serialize both the size of the trees vector and the vector itself.